395 select multiple proposals

dchukhin opened the pull request.

dpoirier commented on symposion/reviews/views.py:
```diff
@@ -93,13 +94,21 @@ def review_section(request, section_slug, assigned=False):
-        proposal = getattr(base_obj, 'pycon%sproposal' % p_type)
-        proposal.overall_status = status
-        proposal.save()
+        pk_string = request.POST['pk']
```
I think I'd rename this input from `pk` to `pks`, both to reflect the change in syntax and to make sure we haven't missed updating any references to the old syntax.

dpoirier commented on symposion/templates/reviews/_review_table.html:
```diff
@@ -132,4 +153,27 @@
+
+<script type="text/javascript">
+$(document).ready(function() {
+    var table = $("table.table-reviews").DataTable();
```
It ought to be possible to put all this new Javascript inside the existing Javascript function (just above this).

dpoirier commented on symposion/reviews/views.py:
```diff
@@ -93,13 +94,21 @@ def review_section(request, section_slug, assigned=False):
+        else:
+            messages.add_message(
+                request, messages.ERROR,
+                (u"Please select at least one application"))
```
This works fine, but there are also shortcut methods for the various message levels that you can use if you prefer, like: `messages.error(request, _("text")`).

dpoirier commented on symposion/reviews/views.py:
```diff
@@ -93,13 +94,21 @@ def review_section(request, section_slug, assigned=False):
+            messages.add_message(
+                request, messages.ERROR,
+                (u"Please select at least one application"))
+            return redirect(request.path)
```
We generally only redirect when there's a successful POST. If there's an error, we just fall through to the regular rendering code.

dpoirier commented on symposion/templates/reviews/_review_table.html:
```diff
@@ -99,15 +99,36 @@
 {% compress js %}
-<script src="{{ STATIC_URL }}datatables/js/jquery.dataTables.min.js" type="text/javascript"></script>
+<script src="{{ STATIC_URL }}datatables/js/jquery-1.11.1.min.js" type="text/javascript"></script>
+<script src="{{ STATIC_URL }}datatables/js/jquery.dataTables-1.10.7.js" type="text/javascript"></script>
```
It seems like something must have been loading datatables already, since this page worked before (though I haven't tried to figure out where).

dchukhin replied:
yeah, something was, but I was not able to call the .rows() method using that file (version 1.9.2); this method seems to have been added later and works with the newer version (1.10.7)

dchukhin replied:
perhaps I can use a minified file

dpoirier commented:
We don't really want to be loading multiple copies of any library.  It could cause hard to figure out bugs where we don't know which version is being called where, as well as using unnecessary bandwidth.

It looks like we're loading jquery from pycon/templates/site_base.html from a  CDN, so if we need a newer version of jquery, we can just update that line to load the newer version.

The datatables javascript is being loaded in individual templates that need them, but it's being loaded from our own copies of the files, so we should be able to just update the versions of those files that we have in our repo and everything will start using the newer versions.

I'm not really happy that we seem to be loading those javascript files for datatables in the middle of the bodies of those pages - I'd rather load them in the header, maybe even from a common parent template - but we don't have to fix that right now.

dchukhin commented:
Ok, I changed the jquery version in pycon/templates/site_base.html to be the 1.11.1 that I was using, and updated the jquery-dataTables.min.js file to be version 1.10.7

dchukhin commented:
We're also not using pycon/static/datatables/js/jquery.js anywhere (and seems like we shouldn't since we're loading it in site_base.html). Should we remove it?

dpoirier commented:
Sure, if we're not loading our own jquery.js, we might as well remove it from the repo.

Trying this locally now, when I select some rows and use the button at the bottom, I get an error "Please select at least one application".

I wonder if we should add some text at the top like "click on a row to select or unselect it", since there's no control visible that would clue people in.

dchukhin commented:
Good thinking about the instructions; I just added some

dpoirier commented:
:+1:
